filters/openpolicyagent/openpolicyagent.go

```diff
@@ -476,25 +477,71 @@ func (registry *OpenPolicyAgentRegistry) new(store storage.Store, configBytes []
 // Start asynchronously starts the policy engine's plugins that download
 // policies, report status, etc.
 func (opa *OpenPolicyAgentInstance) Start(ctx context.Context, timeout time.Duration) error {
-	err := opa.manager.Start(ctx)
+	discoveryPlugin := discovery.Lookup(opa.manager)
+	bundlePlugin := bundle.Lookup(opa.manager)
 
-	if err != nil {
-		return err
+	if discoveryPlugin == nil {
+		return errors.New("discovery plugin not found")
+	}
+	if bundlePlugin == nil {
+		return errors.New("bundle plugin not found")
 	}
 
-	// check readiness of all plugins
-	pluginsReady := func() bool {
-		for _, status := range opa.manager.PluginStatus() {
-			if status != nil && status.State != plugins.StateOK {
-				return false
+	done := make(chan struct{})
+	defer close(done)
+	failed := make(chan error)
+	defer close(failed)
+
+	discoveryPlugin.RegisterListener("startuplistener", func(status bundle.Status) {
+		if len(status.Errors) > 0 {
+			select {
```

Comment:
I don't think the select is needed here (I wouldn't know what it even does in this case), sending to a channel can be done without it.